src/uu/tail/src/args.rs

```diff
@@ -302,34 +354,45 @@ impl Settings {
 
         VerificationResult::Ok
     }
+
+    pub fn is_default(&self) -> bool {
+        let default = Self::default();
+        self.max_unchanged_stats == default.max_unchanged_stats
+            && self.sleep_sec == default.sleep_sec
+            && self.follow == default.follow
+            && self.mode == default.mode
+            && self.pid == default.pid
+            && self.retry == default.retry
+            && self.use_polling == default.use_polling
+            && (self.verbose == default.verbose || self.inputs.len() > 1)
```

Comment:
Maybe I'm being dumb but this doesn't make sense to me. What's your reasoning here? It is the default if there are multiple inputs? Shouldn't the inputs be the same as the default as well? Ideally this would just be a `#[derive(Eq)]`, of course, but maybe that's not possible?

Reply:
Yes. Verbose is automatically added, when there is more than one input:
https://github.com/uutils/coreutils/blob/4e7ae2d46a8550e7bd372a80449662fd646e895e/src/uu/tail/src/args.rs#L221

Basically, the reason is that in calls with obsolete arguments, like `tail +2c`, clap interprets it as "default tail on file `+2c`". My workaround from above, where this is needed, should do something like "if you use the default arguments, please check the first input if it isn't a obsolete argument (instead of a non-existing file (which is not known at this point))".

Therefore Eq is not possible, it would also include the inputs, which are not part of the "settings" (as interpretation of the args).

But I guess I have to add some comments to clarify? :grin:

Reply:
I don't love it to be honest. It seems kind of like a brittle way of doing things.

If it's for the `+` specifically, what about maybe this logic?

1. Try clap
2. If it fails, try obsolete
3. If it succeeds, check the original arguments for whether the length is two and the first character is `+` and try obsolete if that's both true.

Reply:
Yup I find it ugly too. I will look for the cases not covered by clap (e.g. single input with starting +) and run the obsolete parsing on it tomorrow

Reply:
> If it's for the `+` specifically, what about maybe this logic?
>     3. If it succeeds, check the original arguments for whether the length is two and the first character is `+` and try obsolete if that's both true.

Unfortunately not. At least `tail -<num>` is also handled by the obsolete parsing.

Reply:
IMHO, the main problem is to decide ambiguous calls, like e.g. `tail -c +3`:
Parsed by clap, it means "run tail with c=+3 on STDIN".
Parsed by obsolete, it means "run tail with c=-10 on +3".

My latest version checks if clap found only inputs, and if so, runs the obsolete handling afterwards. What do you think?

Reply:
> Unfortunately not. At least tail -<num> is also handled by the obsolete parsing.

Doesn't step 2 take case of that? `-<num>` should fail in clap right?

Reply:
Obviously your solution would also work, and I forgot to implement step 2 yesterday while testing... It's basically the same condition but I think yours is simpler. I'll check later

Reply:
> Obviously your solution would also work

Wasn't so obvious to me :) I'm glad you're doing so much extensive testing